tests/unit/source_code/test_notebook.py

```diff
@@ -82,8 +83,10 @@ def test_notebook_rebuilds_same_code(source: tuple[str, Language, list[str]]):
     assert notebook is not None
     new_source = notebook.to_migrated_code()
     # ignore trailing whitespaces
-    actual_purified = new_source.replace(' \n', '\n')
-    expected_purified = sources[0].replace(' \n', '\n')
+    actual_purified = re.sub(r'\s+$', '', new_source, flags=re.MULTILINE)
+    expected_purified = re.sub(r'\s+$', '', sources[0], flags=re.MULTILINE)
+    if actual_purified != expected_purified:
+        print("what have you done jim?")
```

Comment:
What is this?

Reply:
Accident while debugging. Apologies - I was tired by then.